server/service/src/item/packaging_variant/upsert.rs

```diff
@@ -76,5 +78,35 @@ fn validate(
         return Err(UpsertPackagingVariantError::ItemVariantDoesNotExist);
     }
 
+    let old_packaging_variant =
+        PackagingVariantRowRepository::new(connection).find_one_by_id(&input.id)?;
+    if let Some(old_packaging_variant) = old_packaging_variant {
+        if old_packaging_variant.item_variant_id != input.item_variant_id {
+            return Err(UpsertPackagingVariantError::CantChangeItemVariant);
+        }
+    }
+
+    if input.packaging_level < 0 {
+        return Err(UpsertPackagingVariantError::LessThanZero(
+            "packaging_level".to_string(),
+        ));
+    }
+
+    if let Some(pack_size) = input.pack_size {
+        if pack_size < 0.0 {
```

Comment:
I think this one should be `<=`  - can't have a pack size of 0? Actually, maybe all of these checks should error if the value is 0? Not sure volume of 0, or packaging level of 0 make much sense either...

Reply:
Good point! Done!